### PR TITLE
Handle Rack errors when too many files are uploaded

### PR DESCRIPTION
**Pull request by bschmeck**
Rack [enforces a limit](https://github.com/rack/rack/blob/ed308ff555ae0aeff465044be92bc6bcbc3ddaaa/lib/rack/multipart/parser.rb#L176-L184) on the number of files that can be uploaded in a single request.  Currently, Grape does not handle the error that Rack raises when such a request is received.  This PR catches that error and returns with a `Payload Too Large (413)` status code.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
 * [#2227](https://github.com/ruby-grape/grape/pull/2222): Rename `MissingGroupType` and `UnsupportedGroupType` exceptions - [@ericproulx](https://github.com/ericproulx).
 * [#2244](https://github.com/ruby-grape/grape/pull/2244): Fix a breaking change in `Grape::Validations` provided in 1.6.1 - [@dm1try](https://github.com/dm1try).
 * [#2250](https://github.com/ruby-grape/grape/pull/2250): Add deprecation warning for `UnsupportedGroupTypeError` and `MissingGroupTypeError` - [@ericproulx](https://github.com/ericproulx).
+* [#2256](https://github.com/ruby-grape/grape/pull/2256): Handle MultipartPartLimitError from Rack when too many files are uploaded - [@bschmeck](https://github.com/bschmeck).
```
Comment (dblock) [inline CHANGELOG.md:21]: Let's say "Raise `Grape::Exceptions::TooManyMultipartFiles` on `MultipartPartLimitError`, when too many files are uploaded"?

**Review by dblock — CHANGES_REQUESTED** [review]
Since this does change API behavior, increment the version to 1.7.0 as part of this PR?

**Comment by dblock** [comment]
@bschmeck Yes, please increment the version in version.rb and anywhere else we say "1.6.3", including CHANGELOG

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@
 * [#2227](https://github.com/ruby-grape/grape/pull/2222): Rename `MissingGroupType` and `UnsupportedGroupType` exceptions - [@ericproulx](https://github.com/ericproulx).
 * [#2244](https://github.com/ruby-grape/grape/pull/2244): Fix a breaking change in `Grape::Validations` provided in 1.6.1 - [@dm1try](https://github.com/dm1try).
 * [#2250](https://github.com/ruby-grape/grape/pull/2250): Add deprecation warning for `UnsupportedGroupTypeError` and `MissingGroupTypeError` - [@ericproulx](https://github.com/ericproulx).
+* [#2256](https://github.com/ruby-grape/grape/pull/2256): Raise Grape::Exceptions::MultipartPartLimitError from Rack when too many files are uploaded - [@bschmeck](https://github.com/bschmeck).
```
Comment (dblock) [inline CHANGELOG.md:21]: Back-tick quote `Grape::Exceptions::MultipartPartLimitError`

**Comment by dblock** [comment]
Should/can the message include the system limit number?

**Review by dblock — COMMENTED** [review]
Promise last one: in UPGRADING.md add a paragraph that says something along the lines of 

### Handling Multipart Limit Errors

... if you were handling `Rack::Multipart::MultipartPartLimitError`, Grape will now throw `Grape::Exceptions::TooManyMultipartFiles` ...

**Comment by bschmeck** [comment]
Done.  Thanks for your help.  Let me know if there's anything else!

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Merged, thanks for hanging in here with me @bschmeck !

**Pull request merged**